sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```
@@ -319,13 +348,19 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
      * Note, the parameter is required, or you'll get a warning.
      */
     private static class UnlikeOperator implements Operator {
-        public PlacementRule run(StringMatcher taskFilter, String fieldName, String operatorName,
-                                 Optional<String> requiredParameter) throws IOException {
+        public PlacementRule run(
+                StringMatcher taskFilter,
+                String fieldName,
+                String operatorName,
+                Optional<String> requiredParameter) throws IOException {
+
             String parameter = validateRequiredParameter(operatorName, requiredParameter);
-            if (isHostname(fieldName)) {
-                return HostnameRuleFactory.getInstance().avoid(RegexMatcher.create(parameter));
-            } else {
-                return AttributeRuleFactory.getInstance().avoid(RegexMatcher.createAttribute(fieldName, parameter));
+            switch (PlacementUtils.getField(fieldName)) {
```

should the default case throw an `UnsupportedOperationException`?